### PR TITLE
arc-theme: 20210127 -> 20210412

### PR DESCRIPTION
**Pull request by mkg20001**
###### Motivation for this change

Fix build

###### Things done

Updated src, using meson now

Currently broken as install fails due to mysteriously missing AUTHORS file

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by zimbatm** [comment]
Result of `nixpkgs-review pr 122103` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>arc-theme</li>
  </ul>
</details>

**Review by moni-dz — CHANGES_REQUESTED** [review]
> Currently broken as install fails due to mysteriously missing AUTHORS file

```nix
postInstall = ''
  install -Dm644 -t $out/share/doc/${pname} $src/AUTHORS $src/*.md
'';
```
This should fix the build.

`AUTHORS` and the various Markdown files aren't in the Meson build directory, so we copy them from `src`.

Also, ZHF #122042

**Comment by saschagrunert** [comment]
@mkg20001 let's fixup the build and get this merged :)

**Comment by MetaDark** [comment]
This commit will also need to be rebased against master. The recent rename of gnome -> gnome3: https://github.com/NixOS/nixpkgs/pull/122107 will cause a bad merge that will fail to build.

**Comment by eyJhb** [comment]
> @mkg20001 This commit will also need to be rebased against master. The recent rename of gnome3 -> gnome will result in a successful, but incorrect merge to master that will fail to build. #122107.

Yes, building against the latest unstable-small, gives me the following (pulling this in as a patch).

```
building '/nix/store/6vxgxmbza8yczp2ikid06c0gfi5aj2v2-pkgs-unstable-patched.drv'...
unpacking sources
unpacking source archive /nix/store/iv7g7xdvxh4wjzwn1pv0fvi138kkfy1i-nixpkgs-src
source root is nixpkgs-src
patching sources
applying patch /nix/store/7hswywzaqprvimxlfyzpng7626h8n61r-122103.patch
patching file pkgs/data/themes/arc/default.nix
Hunk #1 succeeded at 1 with fuzz 1.
installing
trace: warning: The option `services.gnome3.gnome-keyring.enable' defined in `/state/home/projects/nixos/nixus-setup/config/machines/eos/work-from-home.nix' has been renamed to `services.gnome.gnome-keyring.enable'.
error: undefined variable 'gnome3' at /nix/store/nx504r7z183adr3slgnkl196j65bkgf5-pkgs-unstable-patched/pkgs/data/themes/arc/default.nix:36:5
(use '--show-trace' to show detailed location information)
```

**Comment by chvp** [comment]
I posted [a commit](https://github.com/chvp/nixpkgs/commit/63a2b6acfb5c204d5ff782948aee4b70bd2b0bc4) in #122086 that fixes the build and works correctly.

**Comment by fortuneteller2k** [comment]
@ofborg eval

**Comment by fortuneteller2k** [comment]
Result of `nixpkgs-review pr 122103` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>arc-theme</li>
  </ul>
</details>

**Review by moni-dz — APPROVED** [review]

File: pkgs/data/themes/arc/default.nix [hunk 1]
```diff
@@ -50,7 +55,7 @@ stdenv.mkDerivation rec {
   ];
```
Comment (chvp) [inline pkgs/data/themes/arc/default.nix:55]: `configureFlags` aren't relevant for a meson build.

**Review by chvp — CHANGES_REQUESTED** [review]

**Comment by fortuneteller2k** [comment]
#122086 was merged already, so it's safe to close this

**Pull request closed without merging**